programs/gateway/src/lib.rs

```diff
@@ -80,6 +130,93 @@ pub mod gateway {
         Ok(())
     }
 
+    /// Withdraws amount to destination program pda, and calls on_call on destination program
+    ///
+    /// # Arguments
+    /// * `ctx` - The instruction context.
+    /// * `amount` - The amount of SOL to withdraw.
+    /// * `sender` - Sender from ZEVM.
+    /// * `data` - Data to pass to destination program.
+    /// * `signature` - The TSS signature.
+    /// * `recovery_id` - The recovery ID for signature verification.
+    /// * `message_hash` - Message hash for signature verification.
+    /// * `nonce` - The current nonce value.
+    pub fn execute(
+        ctx: Context<Execute>,
+        amount: u64,
+        sender: [u8; 20],
+        data: Vec<u8>,
+        signature: [u8; 64],
+        recovery_id: u8,
+        message_hash: [u8; 32],
+        nonce: u64,
+    ) -> Result<()> {
+        let pda = &mut ctx.accounts.pda;
```

Comment:
consider adding verification for amount > 0 , we can probably return early if that is the case